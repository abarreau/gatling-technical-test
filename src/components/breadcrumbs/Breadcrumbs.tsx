import { FC } from 'react';
import styles from './Breadcrumbs.module.css';
import { Link } from 'react-router-dom';

export const Breadcrumbs: FC<{ segments: BreadcrumbSegment[] }> = ({ segments }) => {
    return (
        <div className={styles.breadcrumbsContainer}>
            {
                segments.map((segment, index) => {
                    return index < segments.length - 1
                        ? <span key={index}>
                            <Link className={styles.link} to={segment.url}><span className={styles.link} >{ segment.name }</span></Link>
                            <span className={styles.breadcrumbSeparator}>/</span>
                        </span>
                        : <span key={index}>{ segment.name }</span>;
                })
            }
        </div>
    );
};

export type BreadcrumbSegment = {
    url: string,
    name: string
}