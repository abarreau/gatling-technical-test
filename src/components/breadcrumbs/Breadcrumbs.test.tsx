import renderer from 'react-test-renderer';
import { Breadcrumbs, BreadcrumbSegment } from './Breadcrumbs';
import { MemoryRouter } from 'react-router-dom';

describe('Breadcrumbs', () => {
    it('should display component', () => {
        const segments: BreadcrumbSegment[] = [
            { name: 'Index', url: '/' }, { name: 'Page1', url: '/page1' }
        ];
        const component = renderer.create(
            <MemoryRouter>
                <Breadcrumbs segments={segments}></Breadcrumbs>
            </MemoryRouter>
        );

        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});