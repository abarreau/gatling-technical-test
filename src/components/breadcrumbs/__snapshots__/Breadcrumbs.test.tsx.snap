// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumbs should display component 1`] = `
<div
  className="breadcrumbsContainer"
>
  <span>
    <a
      className="link"
      href="/"
      onClick={[Function]}
    >
      <span
        className="link"
      >
        Index
      </span>
    </a>
    <span
      className="breadcrumbSeparator"
    >
      /
    </span>
  </span>
  <span>
    Page1
  </span>
</div>
`;
